src/components/App/index.tsx

```diff
@@ -51,12 +52,13 @@ const useStyles = makeStyles(notificationStyles)
 const App: React.FC = ({ children }) => {
   const classes = useStyles()
   const { toggleSidebar } = useContext(SafeListSidebarContext)
-  const { name: safeName, totalFiatBalance: currentSafeBalance } = useSelector(currentSafeWithNames)
+  const safesAsMap = useSelector(safesWithNamesAsMap)
   const { safeActionsState, onShow, onHide, showSendFunds, hideSendFunds } = useSafeActions()
   const currentCurrency = useSelector(currentCurrencySelector)
   const granted = useSelector(grantedSelector)
   const sidebarItems = useSidebarItems()
-  const { address: safeAddress } = useSelector(currentSafeWithNames)
+  const safeAddress = extractSafeAddress()
```

Comment:
I thought the issue stemmed from this function? I would be apprehensive to rely on this. It feels as though the Safe may not be rendering when the Safe address in the URL changes. How does a different selector fix this?

Reply:
It would be better to fix the root cause: the selector.

Reply:
`extractSafeAddress` is not a selector. The issue is that we call `extractSafeAddress` inside the `currentSafe` selector which apparently uses a cached value instead of always calling `extractSafeAddress` again. Having the call inside the component ensures that it will always extract the current safe address when rerendering.

Reply:
I know it's not a selector. It's not triggering a rerender because it isn't. I would much prefer we fix the `currentSafe` selector instead of hackily selecting the data we want as a workaround. The current fix is relying on other selectors to re-render the component, hence having the latested value returned from `extractSafeAddress`.